krun_ext_common.py

```diff
@@ -267,9 +267,8 @@ def post_process_dacapo(done_exec):
 
 
 def post_process_spec(done_exec):
-    # We have already check for non-zero exit status, but in addition SPEC
-    # exits with a zero status when a benchmark (not the JVM) fails. Therefore,
-    # check stdout for errors.
+    # We have already checked for non-zero exit status, but SPEC exits zero
```

Comment:
How about "SPEC exits with a zero even if there were problems, so we need to check stdout in order to spot problems."?

Reply:
We need to convey that there's no need to check the return code, as it was checked earlier.

Reply:
I don't see how that's relevant: whether we've checked it or not, by the time we've got to this point we've got check it anyway.

Reply:
> whether we've checked it or not, by the time we've got to this point we've got check it anyway.

No, we don't have to check the return status, only stdout.

There seems to be confusion. We are checking two separate things:
 - The exit status (already checked in the callee)
 - stdout

In case a reader thinks that we have missed the former, I'd like the comment to be explicit.

Reply:
OK, I see. How about this: "Although at this point we've already checked that SPEC exited with zero, that isn't very helpful, because, surprisingly, SPEC always exits with zero, even when there are errors. We have to check stdout in order to spot errors."?

Also, I suspect we shouldn't be calling it "SPEC" but "SPECJVM" (or something like that)? SPEC is the organisation, not the benchmark suite itself.

Reply:
> How about this

That works for me, but it's no shorter than before :P Maybe it's OK to be a long comment if it better describes the nuance?

> Also, I suspect we shouldn't be calling it "SPEC" but "SPECJVM"

Agreed.

Reply:
> Maybe it's OK to be a long comment if it better describes the nuance?

Yeah, it's not so much the length that's the killer, as that the shorter version was easily misread. Sometimes we can only achieve clarity by being a bit longer. Such is life :/